### PR TITLE
Added specific typescript type selector

### LINKED ISSUE #281: "chuck type" leaves trailing colon
For example:

```typescript
const foo: number = 0;
```

Saying "chuck type" leaves the following:

```typescript
const foo: = 0;
```

### PR DESCRIPTION
**Pull request by AndreasArvidsson**
fixes #281

### CODE REVIEW

File: src/languages/typescript.ts [hunk 1]
```diff
@@ -80,6 +86,25 @@ function valueMatcher() {
...
+      : null;
+
+  return {
+    ...simpleSelectionExtractor(editor, node),
+    context: {
+      leadingDelimiterRange,
+    },
+  };
+}
+
```
Comment (pokey) [inline src/languages/typescript.ts:107]: How come the default one didn't work?

**Review by pokey — COMMENTED** [review]
Looks good but please add tests. I'd say at least:

- [x] return type,
- [x] variable declaration type,
- [x] parameter type, and
- [ ] chucking a whole interface.

File: src/languages/typescript.ts (hunk 1 above)
Comment (AndreasArvidsson) [reply src/languages/typescript.ts:107]: The default implementation assumes that the name/key is a previous sibling to the type. That is true for python but not typescript according to the tree sitter.
I have some ideas for rewriting that entire thing to work for both when I have the time.

Comment (pokey) [reply src/languages/typescript.ts:107]: Ah right ok. Fine with having it here for now

**Comment by AndreasArvidsson** [comment]
Tests added

File: src/test/suite/fixtures/recorded/languages/typescript/chuckTypeSoon2.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+spokenForm: chuck type soon
```
Comment (pokey) [inline src/test/suite/fixtures/recorded/languages/typescript/chuckTypeSoon2.yml:1]: Ah this is nice but I actually meant chucking the entire interface, which can also be referred to as "type", just to make sure that wasn't broken here as it has no leading `:`.  But don't remove this test; it's nice too 😊

Comment (AndreasArvidsson) [reply src/test/suite/fixtures/recorded/languages/typescript/chuckTypeSoon2.yml:1]: I didn't know that. Test added.

Is just optional leading delimiter. But always good have tests.

**Review by pokey — APPROVED** [review]
Ok great thanks for fixing this one; been bugging me for a while 😅

**Pull request merged**